lib/new_relic/agent/instrumentation/dynamodb/instrumentation.rb

```diff
@@ -0,0 +1,47 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module Dynamodb
+    PRODUCT = 'DynamoDB'
+    DEFAULT_HOST = 'dynamodb.amazonaws.com'
+
+    def instrument_method_with_new_relic(method_name, *args)
+      return yield unless NewRelic::Agent::Tracer.tracing_enabled?
+
+      NewRelic::Agent.record_instrumentation_invocation(PRODUCT)
+
+      segment = NewRelic::Agent::Tracer.start_datastore_segment(
+        product: PRODUCT,
+        operation: method_name,
+        host: config&.endpoint&.host || DEFAULT_HOST,
+        port_path_or_id: config&.endpoint&.port,
```

Comment:
Did you run into situations where the endpoint/host/port were nil?

Also, where is `config` coming from?

Reply:
config is an object that conveniently already exists inside of the aws client, giving us access to a bunch of info.

Also, no i didn't, it seems like it's always populate, but i added the default host just in case it's possible.

Reply:
Nice! Thanks! I was wondering if we needed the save navigation operators, but since we're relying on an AWS API, I think having the operators will probably be safer for our users if AWS changes things without us knowing.

Reply:
Yeah, for sure, that's totally why i added the & too, just bc it's not ours and we can't for sure know if there are situations it'll be nil, or if something changes in the future